apps/src/gamelab/GameLab.js

```diff
@@ -1020,7 +1020,11 @@ GameLab.prototype.completePreloadIfPreloadComplete = function () {
 GameLab.prototype.onP5Setup = function () {
   if (this.JSInterpreter) {
     // Re-marshal restored preload methods for the interpreter:
-    for (var method in this.gameLabP5.p5._preloadMethods) {
+    const preloadMethods = _.intersection(
+      this.gameLabP5.p5._preloadMethods,
+      this.gameLabP5.getMarshallableP5Properties()
```

Comment:
I don't fully understand why we need to treat preload methods differently?

Reply:
p5 has some wrapping of 'preload' methods to track when async work that has to happen before `setup()` is done.  It's kind of a weird construction, and at one point we exposed some interpreted student code as preload code too (in fact, we may still have this capability).  At this point I'd have to spend some time figuring out why it's all still this way, but a good start would probably be [this commit](https://github.com/code-dot-org/code-dot-org/commit/667602ea9dc4da6502ef63dd1c03525218851baa) which includes some good comments about the wrapping/restoring we have to do with the preload methods.